pgcli/config.py

```diff
@@ -16,3 +16,40 @@ def write_default_config(source, destination, overwrite=False):
         return
 
     shutil.copyfile(source, destination)
+
+def read_config_files(files, sections, keys, error_logger=None):
+    """
+    Reads a list of config files and merges them. The last one will win.
+    :param files: list of files to read
+    :param keys: list of sections to parse
+    :param keys: list of keys to retrieve
+    :param error_logger: callable to log errors
+    :returns: dict, with None for missing keys.
+    """
+    cnf = ConfigObj()
+    for filename in files:
+        try:
+            cnf.merge(ConfigObj(
+                expanduser(filename), interpolation=False))
+        except ConfigObjError as e:
+            if error_logger:
+                assert callable(error_logger)
```

Comment:
An assert only runs if the handler is passed in (not None). I think it makes sense to raise an error if the calling code passed in something that can't be called, because it's the calling code's mistake.